packages/database/test/exp/integration.test.ts

```diff
@@ -97,6 +105,39 @@ describe('Database@exp Tests', () => {
     expect(snap2).to.equal('b');
   });
 
+  // Tests to make sure onValue's data does not get mutated after calling get
```

Comment:
Lets also add tests where the onValue and get are done on paths above and below each other.

Also, maybe a test that checks that nothing strange happens when a listener is removed